packages/frontend/src/pages/index.tsx

```diff
@@ -1,52 +1,352 @@
-import IndexFooter from "@/components/index/IndexFooter";
-import IndexSearchBar from "@/components/index/IndexSearchBar";
-import Head from "next/head";
-import Link from "next/link";
-import { CiMenuBurger } from "react-icons/ci";
-import { useUser } from "@auth0/nextjs-auth0/client";
-import { useEffect } from "react";
+import { useEffect, useMemo, useState, type ReactElement } from "react";
+import SearchBar from "@/components/index/SearchBar";
+import Layout from "@/components/layout/Layout";
+import { type Tag, type Song } from "@prisma/client";
+import SearchFilterTags from "@/components/index/SearchFilterTags";
+import SearchBand from "@/components/index/SearchBand";
+import SearchAlbum from "@/components/index/SearchAlbum";
+import SearchSongList from "@/components/index/SearchSongList";
+import SearchAlbumList from "@/components/index/SearchAlbumList";
+import SearchLoading from "@/components/index/SearchLoading";
+import Image from "next/image";
+
+export type SongType = Song & { tags: Tag[] };
 
 const Home = () => {
-  const { user, error, isLoading } = useUser();
+  const [mounted, setMounted] = useState(false);
+  const [songList, setSongList] = useState<SongType[]>([]);
+  const [filteredSongList, setFilteredSongList] = useState<SongType[]>([]);
+  const [showBand, setShowBand] = useState<boolean | undefined>(false);
+  const [showAlbum, setShowAlbum] = useState<boolean | undefined>(false);
+  const [channelProfile, setChannelProfile] = useState("");
+  const [searchString, setSearchString] = useState("");
+  const [selectedKey, setSelectedKey] = useState<string | null>(null);
+  const [selectedLanguage, setSelectedLanguage] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => setMounted(true), []);
 
   useEffect(() => {
-    console.log(user);
-  }, [isLoading]);
+    if (!mounted) return;
+
+    void (async () => {
+      await fetch("/api/song", {
+        method: "GET",
+      })
+        .then(async (res) => {
+          await res
+            .json()
+            .then((result: SongType[]) => {
+              setSongList(result);
+              setIsLoading(false);
+            })
+            .catch((err) => console.error(err));
+        })
+        .catch((err) => console.error(err));
+    })();
+
+    setSearchString(localStorage.getItem("song-search") ?? "");
+    localStorage.removeItem("song-search");
+  }, [mounted]);
+
+  useEffect(() => {
+    // TODO: use church's account for api key
+    const apiKey = "AIzaSyACcxuHB_5vduPISTHPH5XjJNlZWjSd8R4";
```

Comment:
API keys should **NEVER** be in plain text in committed files, even if its mock, only transfer them in `.env` files

Reply:
✅